src/ClaimerAgent.sol

```diff
@@ -4,76 +4,219 @@ pragma solidity 0.8.17;
+        uint256 remainingPrizes = drawPrizes[computedDrawId].length - nextPrizeIndex;
+        for (uint i = 1; i < remainingPrizes; i++) {
             uint nextClaimFees = env.claimer().computeTotalFees(i);
-            console2.log("nextClaimFees", nextClaimFees);
             if (nextClaimFees - claimFees > claimCostInPrizeTokens) {
-                claimCount = i;
+                targetClaimCount = i;
                 claimFees = nextClaimFees;
             } else {
                 break;
             }
         }
 
-        if (claimCount > 0) {
-            Claim[] memory claims = new Claim[](claimCount);
-            for (uint i = 0; i < claimCount; i++) {
-                claims[i] = draws[drawId].winners[i + draws[drawId].currentClaimIndex];
+        if (targetClaimCount > 0) {
+            console2.log("GO FOR CLAIMS", targetClaimCount);
+        }
+
+        if (targetClaimCount > 0) {
+            // console2.log("claimPrizes(computedDrawId, targetClaimCount)", computedDrawId, targetClaimCount);
+            uint earned = claimPrizes(targetClaimCount);
+            console2.log("CLAIMED ", targetClaimCount, " PRIZES. EARNED ", earned / 1e18);
+            console2.log("\tmin claim fee: ", claimCostInPrizeTokens/1e18);
+        }
+
+    }
+
+    function countContiguousTierPrizes(uint _nextPrizeIndex, uint _claimCount) public view returns (uint8 tier, uint256 count) {
+        uint256 prizeCount;
+        bool init = false;
+        uint8 tier;
+        for (uint p = _nextPrizeIndex; p < _nextPrizeIndex + _claimCount; p++) {
+            if (!init) {
+                tier = drawPrizes[computedDrawId][p].tier;
+                init = true;
+            }
+            if (tier != drawPrizes[computedDrawId][p].tier) {
+                break;
+            }
+            prizeCount++;
+        }
+        return (tier, prizeCount);
+    }
+
+    function countWinners(uint _nextPrizeIndex, uint _claimCount) public view returns (uint256 count) {
+        uint256 winnersLength;
+        address lastWinner;
+        for (uint p = _nextPrizeIndex; p < _nextPrizeIndex + _claimCount; p++) {
+            if (drawPrizes[computedDrawId][p].winner != lastWinner) {
+                winnersLength++;
+                lastWinner = drawPrizes[computedDrawId][p].winner;
+            }
+        }
+        return winnersLength;
+    }
+
+    function countPrizeIndicesPerWinner(
+        uint _nextPrizeIndex,
+        uint _claimCount,
+        uint _winnersLength
+    ) public view returns (uint32[] memory prizeIndexLength) {
+        uint256 winnersLength = countWinners(_nextPrizeIndex, _claimCount);
+        prizeIndexLength = new uint32[](_winnersLength);
+        uint256 winnerIndex;
+        address currentWinner;
+        for (uint p = _nextPrizeIndex; p < _nextPrizeIndex + _claimCount; p++) {
+            if (currentWinner == address(0)) {
+                currentWinner = drawPrizes[computedDrawId][p].winner;
+            }
```

Comment:
Not sure this condition is needed.
For the first iteration, `currentWinner` will be equal to `address(0)` so it should also be different from `drawPrizes[computedDrawId][p].winner`.

Reply:
Trick is that we don't want winnerIndex to increment